stock_lot_filter_available/models/stock_production_lot.py

```diff
@@ -23,7 +24,18 @@ def _search_product_qty(self, operator, value):
         groups = self.env["stock.quant"].read_group(
             domain, ["lot_id", "quantity"], ["lot_id"]
         )
+        involved_lot_ids = [group["lot_id"][0] for group in groups]
+        if (
+            (operator == "=" and value == 0)
+            or (operator == "<" and value > 0)
+            or (operator == "<=" and value >= 0)
+        ):
+            ids.extend(
+                self.env["stock.production.lot"]
+                .search([("id", "not in", involved_lot_ids)])
+                .ids
+            )
         for group in groups:
-            if group["quantity"] > value:
+            if safe_eval("{} {} {}".format(group["quantity"], operator, value)):
```

Comment:
Please don't use `safe_eval`, but use the operators functions and call them:

https://docs.python.org/3/library/operator.html

```python
mapping = {
    ">": operator.gt,
    "<": operator.lt,
    ...
}
mapping[operator](group["quantity"], value)
```